client-reactjs/src/App.js

```diff
@@ -89,93 +117,96 @@ class App extends React.Component {
-                    <PrivateRoute path="/:applicationId/assets/query/:assetId?" component={QueryDetailsForm} />
-                    <PrivateRoute
-                      path="/:applicationId/assets/visualizations/:visualizationId?"
-                      component={VisualizationDetailsForm}
-                    />
-
-                    <PrivateRoute path="/:applicationId/assets" component={Assets} />
-                    <PrivateRoute path="/:applicationId/dataflow/details/:dataflowId?" component={DataflowDetails} />
-                    <PrivateRoute path="/:applicationId/dataflow" component={dataFlowComp} />
-                    <PrivateRoute path="/admin/applications" component={AdminApplications} />
-                    <PrivateRoute path="/admin/bree" component={ScheduledJobsPage} />
-                    <PrivateRoute path="/admin/clusters/:clusterId" component={ClusterDetails} />
-                    <PrivateRoute path="/admin/clusters" component={AdminClusters} />
-                    <PrivateRoute path="/admin/constraints/:tabName?" component={Constraints} />
-                    <PrivateRoute path="/admin/github" component={GitHubSettings} />
-                    <PrivateRoute path="/admin/users" component={Users} />
-                    <PrivateRoute path="/admin/consumers" component={AdminConsumers} />
-                    <PrivateRoute path="/admin/controlsAndRegulations" component={Regulations} />
-                    <PrivateRoute
-                      path="/:applicationId/dataflowinstances/dataflowInstanceDetails/:dataflowId?/:executionGroupId?"
-                      component={DataflowInstanceDetails}
-                    />
-                    <PrivateRoute path="/:applicationId/dataflowinstances" component={DataflowInstances} />
-                    <PrivateRoute path="/:applicationId/actions" component={Actions} />
-                    <PrivateRoute
-                      path="/:applicationId/manualJobDetails/:jobId/:jobExecutionId"
-                      component={ManualJobDetail}
-                    />
-                    {this.props.authWithAzure ? <Route exact path="*" component={getAssets} /> : null}
-                  </Switch>
-                </Suspense>
-              </ErrorBoundary>
-            </Content>
-          </Layout>
-        </Layout>
-      </Router>
+      <ConfigProvider locale={this.locale(this.state.locale)}>
+        <Suspense fallback={null}>
+          <Router history={history}>
+            <Layout className="custom-scroll" style={{ height: '100vh', overflow: 'auto' }}>
+              {this.props.user && this.props.user.token ? (
+                <Header
+                  style={{
+                    backgroundColor: BG_COLOR,
+                    maxHeight: '50px',
+                    position: 'fixed',
+                    zIndex: 100,
+                    width: '100%',
+                  }}>
+                  <AppHeader setLocale={this.setLocale} />
```

Comment:
<AppHeader langButton={<LangSelectButton setLocale={this.setLocale}  />}/>

move all logic to new <LangSelectButton /> component